app/controllers/application_controller.rb

```diff
@@ -1008,10 +1008,17 @@ def add_noindex_header_to_non_canonical
   end
 
   def set_cross_origin_opener_policy_header
+    referrer_host =
+      begin
+        URI(request.referrer).host
```

Comment:
You can use `UrlHelper.relaxed_parse(request.referrer)&.host`, then you don't have to rescue the error yourself.